src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -1220,30 +1213,28 @@ private boolean shouldSectionAtLocation(final Location location)
                                 continue;
                             }
 
-                            // Update end point occurrence to factor in duplicates
-                            final Integer duplicates = duplicateLocations.get(currentLocation);
-                            if (duplicates != null)
-                            {
-                                for (int duplicate = 0; duplicate < duplicates; duplicate++)
-                                {
-                                    nodesToSectionAt.incrementOccurrence(end);
-                                }
-                            }
+                            // Handle start occurrence
+                            final int startOccurrence = nodesToSectionAt
+                                    .getOccurrence(startNode.get()) - 1;
+                            nodesToSectionAt.incrementOccurrence(startNode.get());
+
+                            // Update end point occurrence to factor in any duplicates in the
+                            // polyline
+                            final int duplicates = duplicateLocations.getOrDefault(currentLocation,
+                                    0);
+                            final int endOccurrence = duplicates
+                                    + nodesToSectionAt.getOccurrence(end) - 1;
```

Comment:
elegant! 💯 given the comment explaining the calculation, and the fact that `duplicates` is disposable (single-use), might be best to just do this calculation in one line? i.e. 
```
final int endOccurrence = duplicateLocations.getOrDefault(currentLocation, 0)
    + nodesToSectionAt.getOccurrence(end) - 1;
```

Reply:
Good point, let me update!